src/platform/backend/drm.c

```diff
@@ -239,11 +307,13 @@ add_output(struct gbm_device *device, struct gbm_surface *surface, struct drm_ou
    dsurface->connector = info->connector;
    dsurface->encoder = info->encoder;
    dsurface->crtc = info->crtc;
-   dsurface->surface = surface;
+   dsurface->gbm_surface = surface;
    dsurface->device = device;
 
+   bsurface.use_egldevice = drm.use_egldevice;
+   bsurface.drm_fd = drm.fd;
    bsurface.display = (EGLNativeDisplayType)device;
-   bsurface.display_type = EGL_PLATFORM_GBM_MESA;
+   bsurface.display_type = drm.use_egldevice ? EGL_PLATFORM_DEVICE_EXT : EGL_PLATFORM_GBM_KHR;
```

Comment:
it has to do with priority, I mean here it does not matter, but generally `a ? 3 : 4 +1` is different from `(a ? 3 : 4) + 1`